app/controllers/document_images_controller.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+class DocumentImagesController < ApplicationController
+  def create
+    document = Document.find(params[:id])
+    upload = UploadedImageService.new(params.require(:image)).process
+
+    if upload.valid?
+      image = create_image_from_upload(upload, document)
+      render json: ImageJsonPresenter.new(image).present, status: 201
+    else
+      render json: { errors: upload.errors }, status: 422
+    end
+  end
+
+private
+
+  def create_image_from_upload(upload, document)
```

Comment:
Would it make sense for this to live in the `UploadedImageService`?

Reply:
Potentially - I hadn't thought about putting it there. I did debate putting it into the model but it felt a bit strange to pass the ValidImage models there.

I feel a bit happier with it outside the UploadedImageService as I think it confuses that a bit since it's designed for a single purpose. If we have to move other bits into there though this would be a good candidate.